test/e2e/storage/testsuites/provisioning.go

```diff
@@ -48,6 +48,9 @@ type StorageClassTest struct {
 	ExpectedSize       string
 	PvCheck            func(volume *v1.PersistentVolume) error
 	NodeName           string
+	NodeSelector       map[string]string
+	SecondNodeName     string
```

Comment:
Are there any tests that actually need a second node to be specified?

Reply:
The test that runs two pods runs the second pod on the second node. That way one can test moving a volume from one node to another.

It would be better to have more tests that involve various combinations of multiple pods and multiple nodes, like trying to use the same volume on the same node by two pods, or trying to use the same volume on two different nodes at the same time. But at least a single test is a start, and all of those additional tests would depend on a config option for the node(s) that they should use.

Reply:
This nodeselector was mainly intended for hostpath volumes which can only be scheduled to a specific node. It's also used by multiple tests, not just the provisioning ones.  For plugins like PD, we don't want to set a nodeselector because we want to test the volume scheduling capabilities.

I think the test itself can dynamically construct a node affinity that excludes the node the first pod was scheduled to.  However, something like hostpath or local volumes won't work with this kind of test so it may need to define a plugin capability.

Reply:
For PD it's fine to leave all of these fields empty.

Same with the SecondNodeName: the hostpath driver has to leave it empty for correct execution.

Reply:
I'm not sure if I want to make SecondNodeName a driver configurable parameter.

I'm thinking in the future, we will want a conformance test that explicitly moves the pod to a different node.  For such a conformance test, this second node should be controlled by the test case itself, not an optional argument set by a driver.

Plugins that don't support multi-node, like hostpath or local, will not be able to pass this test and we can add a driver capability to skip it.

Reply:
IMHO such a conformance test can be written without changing `TestConfig`. But when we need to make changes for such a conformance test, we can still make them later, okay?

Reply:
I'm actually thinking we could reuse this same test and promote it as a conformance test.  I don't think we should need to write a separate test just for conformance that does basically the same function as a non-conformance test.  I also don't want to have to remove this field in the future and break people that are vendoring this test suite.

This second node setting, to me, seems more appropriate to drive from the test case itself, and not the driver.  The driver should be able to define its topology constraints, but shouldn't be forcing scheduling behavior.  Setting nodename in the Pod bypasses the scheduler and eliminates that whole path from e2e testing.  I agree that current tests are doing this for hostpath, but I want to refactor the old tests to move away from that model, and move towards a model where we handle topology more holistically.

This new field is adding new functionality to a test and is not part of the refactoring.  I think we should consider enhancing this test case as a separate PR.